packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -955,20 +895,32 @@ export function diffHydratedProperties(
       // satisfies our requirement. Our requirement is not to produce perfect
       // HTML and attributes. Ideally we should preserve structure but it's
       // ok not to if the visible content is still enough to indicate what
-      // even listeners these nodes might be wired up to.
+      // event listeners these nodes might be wired up to.
       // TODO: Warn if there is more than a single textNode as a child.
       // TODO: Should we use domElement.firstChild.nodeValue to compare?
       if (typeof nextProp === 'string') {
         if (domElement.textContent !== nextProp) {
-          if (__DEV__ && !suppressHydrationWarning) {
-            warnForTextDifference(domElement.textContent, nextProp);
+          if (__DEV__) {
```

Comment:
Now the check for `rawProps[SUPPRESS_HYDRATION_WARNING] === true` is performed within `didNotMatchHydratedChildrenPropValue` and similar functions provided by `react-reconciler` `ReactFiberHydrationWarning`, so no need to check it here. Still checking `__DEV__` to ensure this call is not bundled into production build.